controllers/goharbor/harbor/loadblance.go

```diff
@@ -0,0 +1,138 @@
+package harbor
+
+import (
+	"context"
+	"fmt"
+
+	goharborv1alpha2 "github.com/goharbor/harbor-operator/apis/goharbor.io/v1alpha2"
+	harbormetav1 "github.com/goharbor/harbor-operator/apis/meta/v1alpha1"
+	"github.com/goharbor/harbor-operator/controllers"
+	"github.com/goharbor/harbor-operator/pkg/graph"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+type CoreLB graph.Resource
+
+func (r *Reconciler) AddCoreLB(ctx context.Context, harbor *goharborv1alpha2.Harbor, core Core, portal Portal) (CoreLB, error) {
+
+	lbRes, err := r.Controller.AddServiceToManage(ctx, r.GetCoreLB(ctx, harbor), core, portal)
+
+	return CoreLB(lbRes), errors.Wrap(err, "cannot add core lb service")
+}
+
+func (r *Reconciler) GetCoreLB(ctx context.Context, harbor *goharborv1alpha2.Harbor) *corev1.Service {
+	if harbor.Spec.Expose.Core.LoadBalancer == nil {
```

Comment:
harbor.Spec.Expose.Core.LoadBalancer != nil but harbor.Spec.Expose.Core.LoadBalancer.Enable may be false.